library/src/commonMain/kotlin/com/lagradost/cloudstream3/MainAPI.kt

```diff
@@ -1217,6 +1217,28 @@ interface LoadResponse {
         }
     }
 }
+/**
+ * Extracts a four-digit year from a string, prioritizing years in parentheses and ensuring no word characters follow.
+ *
+ * Example:
+ *
+ * "This is (2023) movie" -> 2023
+ *
+ * "This is 1920x1080p movie" -> null
+ *
+ * "This is 2023 movie" -> 2023
+ *
+ * "This is 1999-2010 TvSeries" -> 1999
+ *
+ * @param check The input string.
+ * @return The year as an integer, or `null` if no match is found.
+ */
+fun getYearFromString(check: String?): Int? {
+    return check?.let {
+        "(?<=\\()\\d{4}(?=\\))".toRegex().find(it)?.value?.toIntOrNull()
```

Comment:
Regex construction makes this function take ~10x longer than it should. As such, I would like if you moved both of the .toRegex() into private variables in the global scope